capsules/extra/src/hmac.rs

```diff
@@ -146,10 +146,7 @@ impl<
                                 }
                             })
                         })
-                        .unwrap_or(Err(ErrorCode::RESERVE));
-                    if ret.is_err() {
-                        return ret;
-                    }
+                        .unwrap_or(Err(ErrorCode::RESERVE))?;
```

Comment:
This is a great example. Let's tack a single `?` onto the end of a 25-line block that can invisibly cause an early-return from the function.

Now, realistically, I think I should get over myself and we should do this the 'rust-y' way and make this change. But I will 'old man yell at cloud' and assert that this is an absolutely insane language feature that dramatically harms the readability of Rust code for anyone except full-time Rust developers.

Reply:
FWIW, I think `?` really becomes most clearly useful when it's used really ubiquitously, along-side `Into` implementations between error types. It encourages using meaningful error types (potentially including some state), and can make really long control-flowy code much terser and easier to read, where the control flow is all uniform and easy to reason about---this function is a combinator of a bunch of functions that may or may not succeed, and each of them may return early. Think parser-combinators.

It's a really common pattern in Haskell and is really great when it's used to it's full extent.

All to say, if we start using this _more_ we might get over a hump where there is better local maximum